setup.py

```diff
@@ -2,204 +2,207 @@
+    inc_dirs = [ROOT / "hdf5-blosc" / "src"]
     optional_libs = []
-    data_files = []    # list of data files to add to packages (mainly for DLL's)
+    data_files = []  # list of data files to add to packages (mainly for DLL's)
 
     default_header_dirs = None
     default_library_dirs = None
     default_runtime_dirs = None
 
-
     def add_from_path(envname, dirs):
-        try:
-            dirs.extend(os.environ[envname].split(os.pathsep))
-        except KeyError:
-            pass
-
+        dirs.extend(
+            Path(path)
+            for path in os.environ.get(envname, "").split(os.pathsep)
+            if path
+        )
 
     def add_from_flags(envname, flag_key, dirs):
-        for flag in os.environ.get(envname, "").split():
-            if flag.startswith(flag_key):
-                dirs.append(flag[len(flag_key):])
-
+        dirs.extend(
+            Path(flag[len(flag_key) :])
+            for flag in os.environ.get(envname, "").split()
+            if flag.startswith(flag_key)
+        )
 
-    if os.name == 'posix':
-        prefixes = ('/usr/local', '/sw', '/opt', '/opt/local', '/usr', '/')
+    if os.name == "posix":
+        prefixes = [
+            Path(path)
+            for path in (
+                "/usr/local",
+                "/sw",
+                "/opt",
+                "/opt/local",
+                "/usr",
+                "/",
+            )
+            if Path(path).is_dir()
+        ]
 
         default_header_dirs = []
         add_from_path("CPATH", default_header_dirs)
         add_from_path("C_INCLUDE_PATH", default_header_dirs)
         add_from_flags("CPPFLAGS", "-I", default_header_dirs)
         add_from_flags("CFLAGS", "-I", default_header_dirs)
-        default_header_dirs.extend(
-            os.path.join(_tree, 'include') for _tree in prefixes
-        )
+        default_header_dirs.extend(_tree / "include" for _tree in prefixes)
 
         default_library_dirs = []
         add_from_flags("LDFLAGS", "-L", default_library_dirs)
         default_library_dirs.extend(
-            os.path.join(_tree, _arch) for _tree in prefixes
-            for _arch in ('lib64', 'lib'))
+            _tree / _arch for _tree in prefixes for _arch in ("lib64", "lib")
+        )
         default_runtime_dirs = default_library_dirs
 
-    elif os.name == 'nt':
+    elif os.name == "nt":
         default_header_dirs = []  # no default, must be given explicitly
         default_library_dirs = []  # no default, must be given explicitly
         default_runtime_dirs = [  # look for DLL files in ``%PATH%``
-            _path for _path in os.environ['PATH'].split(';')]
+            Path(_path) for _path in os.environ["PATH"].split(";")
+        ]
         # Add the \Windows\system to the runtime list (necessary for Vista)
-        default_runtime_dirs.append('\\windows\\system')
+        # TODO: should that go to C:\ ?
+        default_runtime_dirs.append(Path("\\windows\\system"))
```

Comment:
I'm not a windows user.
If I remember well I have seen someone with windows installed on ``D:\\``.
Probably the safer option is to stay with the current implementation.

Reply:
As Windows does not have one root dir and this looks like “go up up up and then to windows on this drive” although `PyTables` could be installed somewhere else, I have a bad feeling about this line. I do not want to change it because I do not understand it. But I want to understand it. :-)